diff --git a/scripts/build_backend.py b/scripts/build_backend.py
index a252a1d..96a438c 100755
--- a/scripts/build_backend.py
+++ b/scripts/build_backend.py
@@ -64,6 +64,8 @@ def build_cpp(cc, cxx, sources, targetprefix, targetname, targetsuffix, libs, li
         obj_ext = '.o'
         library_dir_plat = ['-L' + libdir for libdir in libdirs]
         additional_linker_opts = ['-o', f'{targetprefix}{targetname}{targetsuffix}']
+    if IS_LIN:
+        eca += ['--gcc-toolchain=$BUILD_PREFIX/x86_64-conda-linux-gnu']
     eca += ['-c']
     libs = [f'{lib_prefix}{str(item)}{lib_suffix}' for item in libs]
 
