diff --git a/tests/test_examples.py b/tests/test_examples.py
index 98c60a9..43a9100 100755
--- a/tests/test_examples.py
+++ b/tests/test_examples.py
@@ -177,14 +177,6 @@ gen_examples = [
     ('cholesky_batch', 'cholesky_batch.csv', 'choleskyFactor'),
     ('covariance_batch', 'covariance.csv', 'covariance'),
     ('covariance_streaming', 'covariance.csv', 'covariance'),
-    ('decision_forest_classification_default_dense_batch',
-     None, lambda r: r[1].prediction),
-    ('decision_forest_classification_hist_batch',
-     None, lambda r: r[1].prediction, (2021, 'P', 200)),
-    ('decision_forest_regression_default_dense_batch',
-     'decision_forest_regression_batch.csv', lambda r: r[1].prediction, (2021, 'P', 600)),
-    ('decision_forest_regression_hist_batch',
-     'decision_forest_regression_batch.csv', lambda r: r[1].prediction, (2021, 'P', 600)),
     ('decision_tree_classification_batch', 'decision_tree_classification_batch.csv',
      lambda r: r[1].prediction),
     ('decision_tree_regression_batch', 'decision_tree_regression_batch.csv',
